cmd/transcriber/call/transcriber.go

```diff
@@ -40,35 +40,54 @@ type Transcriber struct {
 	captionsPoolDoneCh  chan struct{}
 }
 
-func NewTranscriber(cfg config.CallTranscriberConfig) (*Transcriber, error) {
-	if err := cfg.IsValid(); err != nil {
-		return nil, fmt.Errorf("failed to validate config: %w", err)
+func NewTranscriber(cfg config.CallTranscriberConfig) (t *Transcriber, retErr error) {
+	if err := cfg.IsValidURL(); err != nil {
+		return nil, fmt.Errorf("failed to validate URL: %w", err)
 	}
 
-	client, err := client.New(client.Config{
+	apiClient := model.NewAPIv4Client(cfg.SiteURL)
+	apiClient.SetToken(cfg.AuthToken)
+
+	t = &Transcriber{
+		cfg:       cfg,
+		apiClient: apiClient,
+	}
+
+	defer func() {
+		if retErr != nil {
+			retErrStr := fmt.Errorf("failed to create Transcriber: %w", retErr)
+			if err := t.ReportJobFailure(retErrStr.Error()); err != nil {
+				retErr = fmt.Errorf("failed to report job failure: %s, original error: %s", err.Error(), retErrStr)
+			}
+		}
+	}()
+
+	if retErr = cfg.IsValid(); retErr != nil {
+		return
+	}
```

Comment:
Personal preference but I usually wouldn't explicitly assign the error. You can just return an error as usual.